### PR TITLE
Add web layer component and GitHub Action for deployment automation

### PR DESCRIPTION
**Pull request by mccrackenyyc**
TL;DR: Adds completed web layer to project build along with basic pipeline automation

-----

tfdeploy.yml
- service principal
- updates "env_name" variable to prod on push to main
- executes runner with init and apply
- no pre-testing available, please review that action ran successfully after push, if not then notify me for fix

outputs.tf
- added required output for dns information

variables.tf
- added hostname variable for web hosting

web.tf
- full build on web app layer
- includes geo-redundancy
- built using for_each loops to enable rapid scaling in the future
- test by (post deploy) navigating to: https://gwa-linux-web-app-eus2-dev.azurewebsites.net
- full access to be completed in CDN layer next

### CODE REVIEW

**Review by nexxai — APPROVED** [review]
Yep, everything here looks great.
